### ISSUE TITLE
problems integrating Nim pretty printers

### ISSUE THREAD

**Issue by krux02** [cs01/gdbgui#210]
**Describe the bug**
I have written gdp pretty printers in python for the programming language Nim. When I run gdb from the console, everything runs fine. But when I run gdbgui instead of gdb, my gdbinit is executed. I can see that my pretty printers work. But after the output that was caused by my gdbinit, there are also some error messages.

this is my `.gdbinit`

```
source nim-gdb.py
break bar
run
```

in the screenshot you see the output. As you can see, my breakpoint has been reached successfully, but the the line `set breakpoint pending on` that I marked, does not come from me. And all the error messages later are also not coming from me. Also the stack trace on the right is not from me.

**To Reproduce**
```
git clone git@github.com:krux02/Nim.git
cd Nim
# checkout my feature branch
git checkout macro-in-generic-type
# clone and compile the bootstrapping compiler
git clone --depth 1 https://github.com/nim-lang/csources.git
cd csources
sh build.sh
cd ..
# compile the build tool for the compiler
bin/nim c koch
# compile the real version of the compiler
./koch boot -d:release
# compiler the tiny project that I use to develop the pretty printers
bin/nim c --debugger:native scratch.nim
# start gdbgui on that project
gdbgui scratch
```

**Screenshots**

![screenshot1](https://i.snag.gy/7JuBCF.jpg)

**Please complete the following information:**
* OS: Debian 7.12-6
* gdbgui version: 0.12.0.0:
* gdb version: (7.12.0.20161007-git)
* browser: chrome
* python packages: None

**Additional context**

**Comment by cs01** [comment]
Hi thanks for reporting this.

gdbgui sends commands "behind the scenes" to try to make the experience as good as possible. Sometimes they need some fine tuning. This might be the case here since it looks like you're using nim, which I did not know could be used with gdb (or even existed 😄 ).

Regarding `set breakpoint pending on` that command is sent by gdbgui automatically. There is no option to send/not send it though. I should probably add one to turn it on/off.

I suggest going to settings (gear icon in top right) and unchecking `Add breakpoint to main after loading executable`.

I suspect this breakpoint is added to the address of the main function in a way that gdb could not determine the associated source file, so it puts you in some weird low-level stack at that address and maybe is the source of the "unable to read memory" errors.

See if that helps. If not, go back to settings and check `Print all sent commands in console, including those sent automatically by gdbgui`. Then reproduce all of the spam you get in the console. That will let us see what commands gdbgui is sending and hopefully determine the root cause of the messages.

**Comment by krux02** [comment]
So I did try it out now here at home. So setting the breakpoint for main does not do the effect. But thank you for pointing out that I can disable it. The breakpoint in main isn't very usable becaues that is where the nim runtime is setup. And normally you don't want to debug that. I did find a workaround though.

When I do not have a local .gdbinit, then everything works. But then I also don't have my python script loaded, so I have to type manually 'source nim-gdb.py' inside of gdbgui. I can also hit restart in gdbgui, but then for some reason my python module is unloaded and I have to do 'source nim-gdb.py' as well.

![another screenshot](https://snag.gy/ps6q2t.jpg)

Another issue I have right now. I don't see the string representation of 'NCSTRING'. It just pretends it's a pointer to a single character. But I wrote a pretty printer with the correct output. you can see it in the gdb window.

And the last thing, on the call stack, the arguments to the functions are not listed. This is also gdbgui only, because bt on the terminal works nicely.

**Comment by krux02** [comment]
There is also an issue I have for types where I have the pretty printer overloaded. All of a sudden I can't expand those types anymore. Before I could inspect all the members, now I just can't they are gone.  In the screenshot you see on the console that `print conf.configVars` works nicely. And it works with the pretty printer children iterator. But the configVars field in the tree view is not expandable.

![more screenshot](https://i.snag.gy/H5ispY.jpg)
